### PR TITLE
Fix probe temp calibration

### PR DESCRIPTION
**Pull request by TheNeskik**
Fix 'waiting for printer to cool down' bug

### Requirements

* Having a probe like P.I.N.D.A v2

### Description

This fix allow users to perform probe temperature calibration.
Array values ​​are in the wrong place

### Benefits

Users can perform a G76 without error
### Related Issues

#16711

### CODE REVIEW

**Comment by thisiskeithb** [comment]
Thanks for the fix! I’ll be able to test it later today and I’ll report back.

**Comment by TheNeskik** [comment]
During my tests, I noticed that the probe was using the bed temperature table.

After some tests, everything seems to be good now.

**Comment by thisiskeithb** [comment]
I think this should work despite it failing on my MK3 due to thermal runaway (beta motherboard/settings):

```
Send: G76

X:104.00 Y:101.25 Z:5.00 E:0.00 Count X:10400 Y:10125 Z:2000
Waiting for printer to cool down.

Target Bed: 60; Probe: 30
Waiting for bed and probe to reach target temp.
Waiting for probe heating.

[several bed heating/probe cycles later]

Measured: 0.31
Retrieved measurements: 10
Successfully calibrated bed.

Waiting for bed and probe temperature.

Bed temp: 122; Probe temp: 30  Waiting for probe heating.
Measured: 0.30

[several probe cycles later]

Measured: 0.37
Bed temp: 122; Probe temp: 55  Waiting for probe heating.

Error:Printer halted. kill() called!
```

**Comment by boelle** [comment]
@TheNeskik is there an issue number that this one fix's ?

**Comment by thisiskeithb** [comment]
@boelle: Please read the PR again. It’s there.

**Comment by boelle** [comment]
yeah, the old man was reading to fast

**Comment by tompe-proj** [comment]
Today I tried release 2.0.2 and ran into the same problems. These changes (thanks @TheNeskik) fixed G76 for me. I´m not on the exact same branch but it was just recently merged. Compensation process was successful and M871 worked as expected. I have yet to print something but I just reviewed the corresponding code parts and could not find any other typos. For safety I will debug interpolated values before starting a print tomorrow.

**Comment by TheNeskik** [comment]
@tompe-proj and @thisiskeithb, what kind of heated bed do you have ?, with mine, I am not able to warm up the probe more than 34°C.
Just in case if the calibration needs to be improved :)

**Comment by thisiskeithb** [comment]
> what kind of heated bed do you have ?

Stock (genuine) Prusa MK3 bed, so MK52. The probe got up to 55 before I had a bed thermal runaway error and I haven’t tested or tuned that since.

**Comment by TheNeskik** [comment]
Thanks for your response, I will make some tests soon on other printers.

At the moment, the code seems to be working. I think there needs to be more feedback on the use of this feature.

For example, I had to decrease the measurement temperature of the sensor to 28 ° C because I cannot heat it to 30 ° C with a bed at 60 ° C.

**Comment by tompe-proj** [comment]
I just finished testing, interpolated values are valid. Everything should work as expected now. I'm already printing.

@TheNeskik Not heating up the probe enough is actually the main problem with this. Especially in winter when ambiente temperature is low. Make sure there are no fans cooling it, if needed shield it with e.g. a piece of paper (hotend will not be heated). Also it is important that the probe is as close to the bed as possible, even 1mm makes a huge difference. The procedure places the probe at z=0.5 to heat it up (you could try to change that to 0 in code -> G76_M871.cpp. If you haven't calibrated first layer yet that might be easily 1mm more again. Yesterday I also needed to add my enclosure with door half open as it is currently 18°C in my workshop, in my enclosure I measured 25°C then.

Here is how I did it (MK3S clone):

- Lowering my nozzle until paper between nozzle and bed started scratching.

- Placing a zip-tie on bed and pushing the probe onto it. (Now it should be approximately 1mm higher then nozzle)

- Homing

- Calibrating first layer and setting offset with M851 Z

- Shielding probe from hotend fan and adding enclosure

- Running temperature calibration

I'm using a genuine MK52 (12V) with powder coated steel sheet and genuine PINDA V2.

Tl;dr: Make sure your probe is not cooled by any fan, your room is not too cold, and that it is close enough to the bed (every millimeter counts)

**Comment by ogland** [comment]
I'm still having this issue on 2.0.6. It just sits there at set bed temp forever.
